Documentation/observability/visibility.rst

```diff
@@ -12,63 +12,73 @@ Layer 7 Protocol Visibility
-application without configuring a full policy, Cilium provides a means of
-prescribing visibility via `annotations <https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/>`_
-when running in tandem with Kubernetes.
+While :ref:`monitor` provides introspection into data path state, by default, it
+will only provide visibility into L3/L4 packet events. If you want to get L7
+protocol visibility you can use L7 Cilium Network Policies (see :ref:`l7_policy`).
 
-Visibility information is represented by a comma-separated list of tuples in
-the annotation:
 
-``<{Traffic Direction}/{L4 Port}/{L4 Protocol}/{L7 Protocol}>``
-
-For example:
-
-::
-
-  <Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>
-
-
-To do this, you can provide the annotation in your Kubernetes YAMLs, or via the
-command line, e.g.:
-
-.. code-block:: shell-session
-
-    kubectl annotate pod foo -n bar policy.cilium.io/proxy-visibility="<Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>"
+.. note::
 
-Cilium will pick up that pods have received these annotations, and will
-transparently redirect traffic to the proxy such that the output of
-``cilium monitor`` shows traffic being redirected to the proxy, e.g.:
+    Historically, it had been possible to enable L7 visibility using Pod
+    annotations (``policy.cilium.io/proxy-visibility``). This method is
+    no longer supported and we recommend users to switch to L7 policies instead.
+
+The following example demonstrates how to enable DNS and HTTP visibility for all
+traffic within the ``default`` namespace. We do that by creating a ``CilimNetworkPolicy`` 
+resource containing two L7 rules -- one for DNS and one for HTTP traffic. 
+If all you're interested in is visibility, you can omit or wildcard
+the L7 matching conditions as it is shown below:
+
+.. code-block:: yaml
+
+      apiVersion: "cilium.io/v2"
+      kind: CiliumNetworkPolicy
+      metadata:
+        name: "l7-visibility"
+      spec:
+        endpointSelector:
+          matchLabels:
+            "k8s:io.kubernetes.pod.namespace": default
+        egress:
+        - toPorts:
+          - ports:
+            - port: "53"
+              protocol: ANY
+            rules:
+              dns:
+              - matchPattern: "*"
+        - toEndpoints:
+          - matchLabels:
+              "k8s:io.kubernetes.pod.namespace": default
+          toPorts:
+          - ports:
+            - port: "80"
+              protocol: TCP
+            - port: "8080"
+              protocol: TCP
+            rules:
+              http: [{}]
+        - toEntities:
+          - all
+
+The final rule allowing all egress traffic (``toEntities: [ all ]``) is
+optional and can be removed or replaced with a more specific set of rules.
+
+Based on the above policy, Cilium will pick up all UDP/53, TCP/80 and TCP/8080 
+egress traffic from Pods in the ``default`` namespace and redirect it to the 
+proxy such that the output of ``cilium monitor`` or ``hubble observe`` shows 
```

Comment:
Do we have documentation on the proxy? If so, can you link it here? This would help users who have no idea what the proxy is.